dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layerparam/initializer/PoolingLayerParameterInitializer.java

```diff
@@ -82,20 +93,11 @@ public int getIndex() {
    * @return shape of output
    */
   private int[] computeOutputShape() {
-    final int[] computedShape;
-    switch (inputShape.length) {
-    case 1:
-      computedShape = new int[1];
-      computedShape[0] = (inputShape[0] - kernelHeight) / strideHeight + 1;
-      return computedShape;
-    case 2:
-      computedShape = new int[2];
-      computedShape[0] = (inputShape[0] - kernelHeight) / strideHeight + 1;
-      computedShape[1] = (inputShape[1] - kernelWidth) / strideWidth + 1;
-      return computedShape;
-    default:
-      throw new IllegalArgumentException("Unsupported input dimension: " + Integer.toString(inputShape.length));
-    }
+    final int[] computedShape = new int[2];
+    checkShape();
+    computedShape[0] = (inputShape[0] - kernelHeight) / strideHeight + 1;
```

Comment:
I have a concern that we're missing outputs. For example, suppose that `inputShape[0]` is `6`, `kernelHeight` is `3`, and `strideHeight` is `2`. We compute only for the ranges [0, 2] and [2, 4], but we don't for [4, 5].  I know that the exception will be thrown in `checkShape()` for this case. However, shouldn't we support this? Caffe generates outputs for the range [4, 5]. Thus, the height of Caffe's output is `3`, not `2`.
Sorry that I missed this when reviewing last pull request.